cardano-testnet/src/Test/Runtime.hs

```diff
@@ -97,31 +98,72 @@ data LeadershipSlot = LeadershipSlot
   , slotTime    :: Text
   } deriving (Eq, Show, Generic, FromJSON)
 
-poolNodeToTestnetNode :: PoolNode -> TestnetNode
-poolNodeToTestnetNode PoolNode
-  { poolNodeName
-  , poolNodeSprocket
-  , poolNodeStdinHandle
-  , poolNodeStdout
-  , poolNodeStderr
-  , poolNodeProcessHandle
-  } = TestnetNode
-  { nodeName = poolNodeName
-  , nodeSprocket = poolNodeSprocket
-  , nodeStdinHandle = poolNodeStdinHandle
-  , nodeStdout = poolNodeStdout
-  , nodeStderr = poolNodeStderr
-  , nodeProcessHandle = poolNodeProcessHandle
-  }
+poolNodeStdout :: PoolNode -> FilePath
+poolNodeStdout = nodeStdout . poolRuntime
 
 bftSprockets :: TestnetRuntime -> [Sprocket]
 bftSprockets = fmap nodeSprocket . bftNodes
 
 poolSprockets :: TestnetRuntime -> [Sprocket]
-poolSprockets = fmap poolNodeSprocket . poolNodes
+poolSprockets = fmap (nodeSprocket . poolRuntime) . poolNodes
 
 readNodeLoggingFormat :: String -> Either String NodeLoggingFormat
 readNodeLoggingFormat = \case
   "json" -> Right NodeLoggingFormatAsJson
   "text" -> Right NodeLoggingFormatAsText
   s -> Left $ "Unrecognised node logging format: " <> show s <> ".  Valid options: \"json\", \"text\""
+
+allNodes :: TestnetRuntime -> [NodeRuntime]
+allNodes tr = bftNodes tr <> fmap poolRuntime (poolNodes tr)
+
+-- | Start a node, creating file handles, sockets and temp-dirs.
+startNode
+  :: String
+  -- ^ The tempBaseAbsPath
```

Comment:
Please expand on `tempBaseAbsPath` and `tempAbsPath`. The haddocks don't tell the user anything. Ask @newhoggy if its not immediately obvious what these parameters are doing/generally used for.

Reply:
I totally agree that this should be more clear, but if possible,
I would really like to tackle this in a separate PR.
The test infrastructure uses `Testnet.Conf.Conf` which is a set of FilePaths (mix of relative,
absolute) and other config values.
The arguments to `startNode` all come from `Conf` but I'm reluctant to just pass `Conf` because `Conf` contains other additional data.
Basically `tempBaseAbsPath = takeDirectory tempAbsPath`, so one of the two arguments can be safely removed.

Reply:
:+1: